### PR TITLE
[csprng] Initial port

### PR DESCRIPTION
**Pull request by jherico**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
  - I'm not super clear on when the use of `unofficial::` is mandated.  I modified the original repo based on modern CMake packaging practices and the original repo has been untouched for 6 years.
- [ ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
  - The name is somewhat generic.  I'm open to alternatives for the VCPKG portname.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

I'm working on bringing cesium-native into vcpkg, and this is one of its upstream dependencies which isn't currently present.

### CODE REVIEW

**Comment by dg0yt** [comment]
> I'm not super clear on when the use of `unofficial::` is mandated.  I modified the original repo based on modern CMake packaging practices and the original repo has been untouched for 6 years.

It is mandated when it doesn't come from upstream.
Here, it doesn't come from upstream.
So you must use `unofficial`.

File: ports/csprng/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO Duthomhas/CSPRNG
```
Comment (dg0yt) [inline ports/csprng/portfile.cmake:5]: Original upstream

File: ports/csprng/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "csprng",
+  "version": "1.0.0",
+  "description": "OS-native Cryptographically-Secure Pseudo-Random Number Generator wrapper",
+  "homepage": "https://github.com/jherico/CSPRNG",
```
Comment (dg0yt) [inline ports/csprng/vcpkg.json:5]: Fork

**Comment by Adela0814** [comment]
Pinging @jherico  for response. Is work still being done for this PR?

**Pull request closed without merging**